### PR TITLE
Release 2.0 release scripts changes

### PR DESCRIPTION
**Pull request by atalman**
Release 2.0 release scripts changes

1. Add copy dependencies for whl s3
2. Set new versions
3. Add torchdata promotion. Should we do it or let Torchdata team handle it ? I think it will make sence to do it all by us and rerun validation workflows. after this

Still pending:
1. Copy dependencies for conda
2. pipy torchdata promotion. This one perhaps could still be done by torchdata team ?

### CODE REVIEW

File: release/promote.sh [hunk 1]
```diff
@@ -70,16 +71,46 @@ promote_pypi() {
...
+promote_s3 "charset_normalizer" whl "2.1.1"
+promote_s3 "cmake" whl "3.25"
+promote_s3 "colorama" whl "0.4.6"
+promote_s3 "triton" whl "2.0.0"
+promote_s3 "pytorch_triton_rocm" whl "2.0.0"
+promote_s3 "tqdm" whl "4.64.1"
+promote_s3 "Pillow" whl "9.3.0"
+# for python 3.8-3.11
+promote_s3 "numpy" whl "1.24.1"
+# for python 3.7 - should we remove this dependency ?
```
Comment (malfet) [inline release/promote.sh:85]: It's still needed for previous versions

File: release/promote.sh [hunk 2]
```diff
@@ -70,16 +71,46 @@ promote_pypi() {
     echo
 }
 
+# Promote dependencies
+promote_s3 "certifi" whl "2022.12.7"
```
Comment (seemethere) [inline release/promote.sh:75]: This is not my favorite thing in the world but security right?

**Review by seemethere — APPROVED** [review]

File: release/promote.sh [hunk 3]
```diff
@@ -70,16 +71,46 @@ promote_pypi() {
...
+promote_s3 "lit" whl "15.0.7"
+# why sympy has 2 depencencies ? todo: comment here
+promote_s3 "sympy" whl "1.11.1"
+promote_s3 "sympy" whl "1.10.1"
+promote_s3 "typing_extensions" whl "4.4.0"
+promote_s3 "filelock" whl "3.9.0"
+promote_s3 "mpmath" whl "1.2.1"
+promote_s3 "idna" whl "3.4"
+# why networkx 2 depencencies ? todo: comment here
+promote_s3 "networkx" whl "3.0rc1"
```
Comment (malfet) [inline release/promote.sh:97]: Please delete 3.0rc1, migrate to 3.0 (released on Jan 7th) and delete 2.6.3

File: release/promote.sh [hunk 4]
```diff
@@ -70,16 +71,46 @@ promote_pypi() {
...
+promote_s3 "Pillow" whl "9.3.0"
+# for python 3.8-3.11
+promote_s3 "numpy" whl "1.24.1"
+# for python 3.7 - should we remove this dependency ?
+promote_s3 "numpy" whl "1.21.6"
+promote_s3 "urllib3" whl "1.26.13"
+promote_s3 "lit" whl "15.0.7"
+# why sympy has 2 depencencies ? todo: comment here
+promote_s3 "sympy" whl "1.11.1"
+promote_s3 "sympy" whl "1.10.1"
```
Comment (malfet) [inline release/promote.sh:91]: Please delete 1.10.1

**Review by malfet — APPROVED** [review]

File: release/promote.sh (hunk 2 above)
Comment (atalman) [reply release/promote.sh:75]: Yes will have to add it to runbook if new versions of dependencies are uploaded we should do the change here

**Comment by johnnynunez** [comment]
Ooh! so much to look forward to!

**Pull request merged**